oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -172,6 +172,7 @@ private FlatFileStore buildFlatFileStore(NodeState checkpointedState, CompositeI
                         .withBlobStore(indexHelper.getGCBlobStore())
                         .withPreferredPathElements((preferredPathElements != null) ? preferredPathElements : indexer.getRelativeIndexedNodeNames())
                         .addExistingDataDumpDir(indexerSupport.getExistingDataDumpDir())
+                        .withPathPredicate((pathPredicate != null) ? pathPredicate : indexer::shouldInclude)
```

Comment:
We have removed the check for hidden nodes (NodeStateUtils.isHiddenPath), but I don't see where we add the check in the higher level (that is: here), in case there is a pathPredicate... Does pathPredicate do a check for hidden nodes? If yes, why did we check for both pathPredicate.test(path) AND hidden nodes before? (Maybe it's correct, but possibly not).

Reply:
removed and refactored

check is in TraverseAndSortTask